Looks like a bunch of PRs are failing CI with an infra issue. Let me reach out to some folks.

```C#
D:\a\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error : Response status code does not indicate success: 503 (Service Unavailable). [D:\a\1\s\eng\sendtohelix.proj]
D:\a\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error :    at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode() [D:\a\1\s\eng\sendtohelix.proj]
D:\a\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error :    at Microsoft.DotNet.Helix.AzureDevOps.StopAzurePipelinesTestRun.<>c__DisplayClass4_0.<<ExecuteCoreAsync>b__0>d.MoveNext() in /_/src/Microsoft.DotNet.Helix/Sdk/StopAzurePipelinesTestRun.cs:line 36 [D:\a\1\s\eng\sendtohelix.proj]
D:\a\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error : --- End of stack trace from previous location where exception was thrown --- [D:\a\1\s\eng\sendtohelix.proj]
D:\a\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error :    at Microsoft.DotNet.Helix.AzureDevOps.AzureDevOpsTask.<>c__DisplayClass13_0.<<RetryAsync>b__0>d.MoveNext() in /_/src/Microsoft.DotNet.Helix/Sdk/AzureDevOpsTask.cs:line 87 [D:\a\1\s\eng\sendtohelix.proj]
D:\a\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error : --- End of stack trace from previous location where exception was thrown --- [D:\a\1\s\eng\sendtohelix.proj]
D:\a\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error :    at Microsoft.DotNet.Helix.Client.HelixApi.RetryAsync[T](Func`1 function, Action`1 logRetry, Func`2 isRetryable, CancellationToken cancellationToken) in /_/src/Microsoft.DotNet.Helix/Client/CSharp/HelixApi.cs:line 76 [D:\a\1\s\eng\sendtohelix.proj]
D:\a\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error :    at Microsoft.DotNet.Helix.AzureDevOps.StopAzurePipelinesTestRun.ExecuteCoreAsync(HttpClient client) in /_/src/Microsoft.DotNet.Helix/Sdk/StopAzurePipelinesTestRun.cs:line 17 [D:\a\1\s\eng\sendtohelix.proj]
D:\a\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error :    at Microsoft.DotNet.Helix.AzureDevOps.AzureDevOpsTask.Execute() in /_/src/Microsoft.DotNet.Helix/Sdk/AzureDevOpsTask.cs:line 67 [D:\a\1\s\eng\sendtohelix.proj]
##[error].packages\microsoft.dotnet.helix.sdk\5.0.0-beta.19510.4\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(37,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Response status code does not indicate success: 503 (Service Unavailable).
   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at Microsoft.DotNet.Helix.AzureDevOps.StopAzurePipelinesTestRun.<>c__DisplayClass4_0.<<ExecuteCoreAsync>b__0>d.MoveNext() in /_/src/Microsoft.DotNet.Helix/Sdk/StopAzurePipelinesTestRun.cs:line 36
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.DotNet.Helix.AzureDevOps.AzureDevOpsTask.<>c__DisplayClass13_0.<<RetryAsync>b__0>d.MoveNext() in /_/src/Microsoft.DotNet.Helix/Sdk/AzureDevOpsTask.cs:line 87
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.DotNet.Helix.Client.HelixApi.RetryAsync[T](Func`1 function, Action`1 logRetry, Func`2 isRetryable, CancellationToken cancellationToken) in /_/src/Microsoft.DotNet.Helix/Client/CSharp/HelixApi.cs:line 76
   at Microsoft.DotNet.Helix.AzureDevOps.StopAzurePipelinesTestRun.ExecuteCoreAsync(HttpClient client) in /_/src/Microsoft.DotNet.Helix/Sdk/StopAzurePipelinesTestRun.cs:line 17
   at Microsoft.DotNet.Helix.AzureDevOps.AzureDevOpsTask.Execute() in /_/src/Microsoft.DotNet.Helix/Sdk/AzureDevOpsTask.cs:line 67

Build FAILED.
```

--- a/src/System.Text.Json/tests/Serialization/Value.ReadTests.cs
+++ b/src/System.Text.Json/tests/Serialization/Value.ReadTests.cs
@@ -383,20 +383,25 @@ namespace System.Text.Json.Serialization.Tests
             DeserializeLongJsonString(length);
         }
 
+        private const int MaxInt = int.MaxValue / MaxExpansionFactorWhileTranscoding;
+
         // NOTE: VeryLongInputString test is constrained to run on Windows and MacOSX because it causes
         //       problems on Linux due to the way deferred memory allocation works. On Linux, the allocation can
         //       succeed even if there is not enough memory but then the test may get killed by the OOM killer at the
         //       time the memory is accessed which triggers the full memory allocation.
         [ConditionalTheory(nameof(IsX64))]
         [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
-        [InlineData(int.MaxValue - 3)]
-        [InlineData(int.MaxValue - 2)]
-        [InlineData(int.MaxValue - 1)]
-        [InlineData(int.MaxValue)]
+        [InlineData(MaxInt - 3)]
+        [InlineData(MaxInt - 2)]
+        [InlineData(MaxInt - 1)]
+        [InlineData(MaxInt)]
+        [InlineData(MaxInt + 1)]
+        [InlineData(MaxInt + 2)]
+        [InlineData(MaxInt + 3)]
         [OuterLoop]
         public static void VeryLongInputString(int length)
         {
-            // Verify that deserializer does not do any multiplication or addition on the length
+            // Verify that deserializer does not do any multiplication on the length
             DeserializeLongJsonString(length);
         }
 

--- a/src/System.Text.Json/tests/Serialization/Value.ReadTests.cs
+++ b/src/System.Text.Json/tests/Serialization/Value.ReadTests.cs
@@ -384,7 +384,7 @@ namespace System.Text.Json.Serialization.Tests
         }
 
         private const int MaxInt = int.MaxValue / MaxExpansionFactorWhileTranscoding;
-        private const int MaximumPossibleStringLength = (int.MaxValue / 2) - 32;
+        private const int MaximumPossibleStringLength = int.MaxValue / 2 - 32;
 
         // NOTE: VeryLongInputString test is constrained to run on Windows and MacOSX because it causes
         //       problems on Linux due to the way deferred memory allocation works. On Linux, the allocation can
@@ -412,12 +412,30 @@ namespace System.Text.Json.Serialization.Tests
 
         private static void DeserializeLongJsonString(int stringLength)
         {
-            string repeated = new string('x', stringLength - 2);
-            string json = $"\"{repeated}\"";
+            string json, str;
+
+#if BUILDING_INBOX_LIBRARY
+            char fillChar = 'x';
+            json = string.Create(stringLength, fillChar, (chars, fillChar) =>
+            {
+                chars.Fill(fillChar);
+                chars[0] = '"';
+                chars[chars.Length - 1] = '"';
+            });
+
             Assert.Equal(stringLength, json.Length);
 
-            string str = JsonSerializer.Deserialize<string>(json);
+            str = JsonSerializer.Deserialize<string>(json);
+            Assert.True(json.AsSpan(1, json.Length - 2).SequenceEqual(str));
+#else
+            string repeated = new string('x', stringLength - 2);
+            json = $"\"{repeated}\"";
+
+            Assert.Equal(stringLength, json.Length);
+
+            str = JsonSerializer.Deserialize<string>(json);
             Assert.Equal(repeated, str);
+#endif
         }
     }
 }

--- a/src/System.Text.Json/tests/Serialization/Value.ReadTests.cs
+++ b/src/System.Text.Json/tests/Serialization/Value.ReadTests.cs
@@ -384,6 +384,7 @@ namespace System.Text.Json.Serialization.Tests
         }
 
         private const int MaxInt = int.MaxValue / MaxExpansionFactorWhileTranscoding;
+        private const int MaximumPossibleStringLength = (int.MaxValue / 2) - 32;
 
         // NOTE: VeryLongInputString test is constrained to run on Windows and MacOSX because it causes
         //       problems on Linux due to the way deferred memory allocation works. On Linux, the allocation can
@@ -398,10 +399,14 @@ namespace System.Text.Json.Serialization.Tests
         [InlineData(MaxInt + 1)]
         [InlineData(MaxInt + 2)]
         [InlineData(MaxInt + 3)]
+        [InlineData(MaximumPossibleStringLength - 3)]
+        [InlineData(MaximumPossibleStringLength - 2)]
+        [InlineData(MaximumPossibleStringLength - 1)]
+        [InlineData(MaximumPossibleStringLength)]
         [OuterLoop]
         public static void VeryLongInputString(int length)
         {
-            // Verify that deserializer does not do any multiplication on the length
+            // Verify that deserializer does not do any multiplication or addition on the string length
             DeserializeLongJsonString(length);
         }
 

